client/basic-authorship/src/basic_authorship.rs

```diff
@@ -210,7 +210,18 @@ impl<A, B, Block, C> ProposerInner<B, Block, C, A>
 		let mut is_first = true;
 		let mut skipped = 0;
 		let mut unqueue_invalid = Vec::new();
-		let pending_iterator = self.transaction_pool.ready();
+		let pending_iterator = match executor::block_on(future::select(
+			self.transaction_pool.ready_at(Some(self.parent_number)),
+			futures_timer::Delay::new((deadline - (self.now)()) / 8),
```

Comment:
Hmm this deadline seem really random.

Did you test this with a `dev` chain node and tx spamming?

Reply:
See [discussion](https://github.com/paritytech/substrate/pull/5227#discussion_r391520954) above with @tomusdrw.

This timeout is merely avoid thread locking if something is wrong with the pipeline. The future itself supposed to resolve almost instantly (or made to do so if not). If you have ideas what would be less random for this cases, pls share.

I always test prs

Reply:
I don't wanted to say that you don't test stuff :D I just don't know how long it takes until the transaction pool is ready after producing a block? However, I would assume that with tx spamming on a dev-chain we see some problems (skipping blocks), but if you say that everything is fine, I'm happy :)

Reply:
Everything is fine :)